src/blockchain/examples/simple_node.rs

```diff
@@ -18,10 +18,255 @@ extern crate pretty_env_logger;
 extern crate pyrsia_blockchain_network;
 extern crate tokio;
 
-use pyrsia_blockchain_network::Example;
+use futures::StreamExt;
+use libp2p::{
+    core::upgrade,
+    floodsub::{self, Floodsub, FloodsubEvent},
+    identity,
+    mdns::{Mdns, MdnsEvent},
+    mplex,
+    noise,
+    swarm::{NetworkBehaviourEventProcess, SwarmBuilder, SwarmEvent},
+    // `TokioTcpConfig` is available through the `tcp-tokio` feature.
+    tcp::TokioTcpConfig,
+    Multiaddr,
+    NetworkBehaviour,
+    PeerId,
+    Transport,
+};
+use rand::Rng;
+use std::error::Error;
+use tokio::io::{self, AsyncBufReadExt};
 
+pub const BLOCK_FILE_PATH: &str = "./first";
+pub const CONTINUE_COMMIT: &str = "1";
+pub const APART_ONE_COMMIT: &str = "2";
+
+/// The `tokio::main` attribute sets up a tokio runtime.
 #[tokio::main]
-async fn main() {
-    pretty_env_logger::init();
-    let _e: Example = 0;
+async fn main() -> Result<(), Box<dyn Error>> {
+    // Create a random PeerId
+    let id_keys = pyrsia_blockchain_network::blockchain::generate_ed25519();
+    let peer_id = PeerId::from(id_keys.public());
+    println!("Local peer id: {:?}", peer_id);
+    let filepath = match std::env::args().nth(1) {
+        Some(v) => v,
+        None => String::from(BLOCK_FILE_PATH),
+    };
+
+    // Create a keypair for authenticated encryption of the transport.
+    let noise_keys = noise::Keypair::<noise::X25519Spec>::new()
+        .into_authentic(&id_keys)
+        .expect("Signing libp2p-noise static DH keypair failed.");
+
+    // Create a tokio-based TCP transport use noise for authenticated
+    // encryption and Mplex for multiplexing of substreams on a TCP stream.
+    let transport = TokioTcpConfig::new()
+        .nodelay(true)
+        .upgrade(upgrade::Version::V1)
+        .authenticate(noise::NoiseConfig::xx(noise_keys).into_authenticated())
+        .multiplex(mplex::MplexConfig::new())
+        .boxed();
+
+    // Create a Floodsub topic
+    let floodsub_topic = floodsub::Topic::new("block");
+
+    // We create a custom network behaviour that combines floodsub and mDNS.
+    // The derive generates a delegating `NetworkBehaviour` impl which in turn
+    // requires the implementations of `NetworkBehaviourEventProcess` for
+    // the events of each behaviour.
+    #[derive(NetworkBehaviour)]
+    #[behaviour(event_process = true)]
+    struct MyBehaviour {
```

Comment:
I'd like to give this a better name, maybe

```suggestion
    struct BlockchainNetworkBehaviour {
```

Reply:
```suggestion
    # TODO(prince-chrismc): Give this a much better name!
    struct MyBehaviour {
```